### PR TITLE
adapt NostrSDK to vcpkg build system

### PR DESCRIPTION
**Pull request by finrodfelagund97**
@buttercat1791 
As you will see, I've scarcely changed anything other than adapt CMakeLists.txt to the more global build of `GitRepublic`

Please clone GitRepublic and work from within it after you approve of this PR.

### CODE REVIEW

File: src/nostr_service.cpp [hunk 1]
```diff
@@ -517,8 +513,9 @@ void NostrService::disconnect(string relay)
 
 string NostrService::generateSubscriptionId()
 {
-    uuid uuid = random_generator()();
-    return to_string(uuid);
+    UUIDv4::UUIDGenerator<std::mt19937_64> uuidGenerator;
+    UUIDv4::UUID uuid = uuidGenerator.getUUID();
+    return uuid.bytes();
```
Comment (buttercat1791) [inline src/nostr_service.cpp]: Can we add `using UUIDv4::UUIDGenerator` and `using UUIDv4::UUID` to the usings block at the top of this file?  That will keep the code a bit more readable in-line, IMO.

File: CMakeLists.txt [hunk 2]
```diff
@@ -27,21 +35,20 @@ set(SOURCES
...
 find_package(nlohmann_json CONFIG REQUIRED)
+find_package(uuid_v4 REQUIRED)
 find_package(OpenSSL REQUIRED)
 find_package(plog CONFIG REQUIRED)
 find_package(websocketpp CONFIG REQUIRED)
 
 add_library(NostrSDK ${SOURCES} ${HEADERS})
 target_link_libraries(NostrSDK PRIVATE
-    Boost::random
-    Boost::system
```
Comment (buttercat1791) [inline CMakeLists.txt:39]: Stating for the record that we _may_ need to re-add Boost at a later date.  I believe `websocketpp` uses it as a dependency, but we don't test the `websocketpp` usage directly, as of yet.

**Review by buttercat1791 — APPROVED** [review]
Looks good overall.  I left two minor comments.  Please take a look at those, then feel free to merge when you're ready.

File: CMakeLists.txt (hunk 2 above)
Comment (finrodfelagund97) [reply CMakeLists.txt:39]: You are right, websocketpp uses it as a dependency but vcpkg downloads the required parts of boost (already compiled) for websocketpp

But if we directly use boost, vcpkg will have to download all of it fr and compile it from source which took me around 2hrs, and all that just for generating UUIDs... it's a lot of overhead in my opinion.  

If we need it in the future, we'll cross that bridge when we get to it.

File: src/nostr_service.cpp (hunk 1 above)
Comment (finrodfelagund97) [reply src/nostr_service.cpp]: Done in commit
https://github.com/buttercat1791/NostrSDK/pull/4/commits/2c9956cc35644a83f8035cb14302fbfd27070d9b

**Pull request merged**